### ISSUE TITLE
Can not reproduce the result of ICP(97%)

### ISSUE THREAD

**Issue by CodeLHY** [ethnhe/FFB6D#55]
@ethnhe  This is really a nice work, however, when I tried to reproduce the result of FFB6D+ICP using the ICP code from PVN3D I can not reproduce the result, can you give me a hand?

**Comment by CodeLHY** [comment]
https://github.com/ethnhe/PVN3D/blob/9a5481a82af4eb89fa8e0ab10ba22b9077b4652e/pvn3d/eval_icp.py#L124
It seems that you are using the ground truth mask for ICP refinement.

**Comment by ethnhe** [comment]
No, as illustrated in A.3. Implementation: PVN3D+ICP in our paper, we use the post-processed predicted mask for the whole images, using the fill label function. https://github.com/ethnhe/PVN3D/blob/9a5481a82af4eb89fa8e0ab10ba22b9077b4652e/pvn3d/eval_icp.py#L429
This procedure is slow so we pre-process and store it in a data structure. We use CPU clusters to speed up.

**Comment by CodeLHY** [comment]
> No, as illustrated in A.3. Implementation: PVN3D+ICP in our paper, we use the post-processed predicted mask for the whole images, using the fill label function. https://github.com/ethnhe/PVN3D/blob/9a5481a82af4eb89fa8e0ab10ba22b9077b4652e/pvn3d/eval_icp.py#L429 This procedure is slow so we pre-process and store it in a data structure. We use CPU clusters to speed up.

Thank you for your quick reply, and I really appreciate PVN3D and FFB6D, both of them are excellent and solid. Good night.

**Comment by CodeLHY** [comment]
Hi @ethnhe , I followed your instruction but still get a lower result, so I reopened this issue.
Average of all object:
***************add:      87.39481105187926
***************adds:     95.32809689468347
***************add(-s):  91.22332797027117
All object (following PoseCNN):
***************add:      87.82876641272729
***************adds:     95.32953369790825
***************add(-s):  90.4528140615951

could you please provide the files needed for eval_icp.py?

**Comment by Dovisya** [comment]
> Hi @ethnhe , I followed your instruction but still get a lower result, so I reopened this issue. Average of all object: ***************add: 87.39481105187926 ***************adds: 95.32809689468347 ***************add(-s): 91.22332797027117 All object (following PoseCNN): ***************add: 87.82876641272729 ***************adds: 95.32953369790825 ***************add(-s): 90.4528140615951
> 
> could you please provide the files needed for eval_icp.py?

I also used the icp method in pvn3d, combined with the ffb6d code. However, the results of the test on ycb data were reduced. Have you solved this problem? Hope to get your reply, we can communicate with each other

**Comment by CoolCode4** [comment]
Hi @ethnhe and @CodeLHY,

I am also attempting to reproduce the FFB6D+ICP results on YCB-Video, but like @Dovisya, my performance decreased after adding ICP.

In my implementation:

I used the predicted mask from FFB6D (refined by MeanShift clustering) to extract the scene point cloud, not the GT mask.

I applied a MeanShift filter to the scene point cloud to remove outliers before running ICP.

My ICP parameters: max_iter=100, tolerance=1e-6, and a minimum point requirement of 400.

Despite this, the ADD-S AUC is still significantly lower than the paper's 97.0%.

Could you please clarify:

Point Sampling: How many points do you sample from the predicted mask and the model mesh for ICP?

Mask Post-processing: Besides fill label, do you use any specific morphological operations (dilation/erosion) to refine the mask edges?

ICP Hyperparameters: What are the exact search_radius or distance_threshold values used in your ICP configuration to avoid "drifting"?

Any guidance or a snippet of the ICP configuration would be greatly appreciated. Thanks!
